notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "8a6c7193-0499-4f89-b6c8-b0e6a1634149",
+   "metadata": {},
+   "source": [
+    "# Using Joins effectively in Druid \n",
+    "<!--\n",
+    "  ~ Licensed to the Apache Software Foundation (ASF) under one\n",
+    "  ~ or more contributor license agreements.  See the NOTICE file\n",
+    "  ~ distributed with this work for additional information\n",
+    "  ~ regarding copyright ownership.  The ASF licenses this file\n",
+    "  ~ to you under the Apache License, Version 2.0 (the\n",
+    "  ~ \"License\"); you may not use this file except in compliance\n",
+    "  ~ with the License.  You may obtain a copy of the License at\n",
+    "  ~\n",
+    "  ~   http://www.apache.org/licenses/LICENSE-2.0\n",
+    "  ~\n",
+    "  ~ Unless required by applicable law or agreed to in writing,\n",
+    "  ~ software distributed under the License is distributed on an\n",
+    "  ~ \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+    "  ~ KIND, either express or implied.  See the License for the\n",
+    "  ~ specific language governing permissions and limitations\n",
+    "  ~ under the License.\n",
+    "  -->\n",
+    "\n",
+    "Apache Druid can do joins at query time through both its native engine (scatter/gather) and through the Multi-Stage Query engine.\n",
+    "Note that the best performance will always be without using any joins, so the overall recommendation is to join the data at ingestion when loading in batch, or upstream of ingestion for streaming scenarios. But pre-joining the data isn't always an option, so you'll need to learn how to use Druid to process joins as efficiently as possible for those cases.\n",
```

Comment:
I'd rephrase this as a Tell instead:

"Through this notebook you will run tests on the performance of different approaches to joining data, seeing how different JOIN operations are processed and how to write SQL that... blah"

Reply:
I think of this notebook as a one-stop-shop for understanding joins in Druid. I agree that we do not want to provide specific advice on a given situation but we do want to teach what the different Join tools are and what they each do internally, such that developers and data engineers have the tools to design and test different query patterns to make better modeling decisions. I'll reread to make sure, I'm not saying you should use X or Y, but rather, "the tools are X,Y,Z, X works like this, Y works like that, Z works in this other way". 
As an app developer in the past, having this kind of information is extremely valuable when making data architecture decisions that take advantage of the platform.